core/libp2p/common/peer_info.hpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_PEER_INFO_HPP
+#define KAGOME_PEER_INFO_HPP
+
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/multi/multihash.hpp"
+
+namespace libp2p::common {
+  /**
+   * Information about the peer in the network
+   */
+  struct PeerInfo {
```

Comment:
Is there any point for splitting PeerInfo, Peer, NetworkMessage on separate files? I'm pretty sure that they might depend on each other

Reply:
There is no circular dependency, so it is better to keep them separate

Reply:
1. I'm pretty sure that these dependencies will be required in together almost always
2. It's a common base structure, so much easier to look at them without switching the files
3. Structs are empty yet, there's **still** no dependency, but might be

Reply:
I'm pretty sure that `Peer` may contain its `PeerInfo` as well as `NetworkMessage` may contain a `PeerInfo`, but still they are separate entities, which may (and will) exist alone in the system. For example, to establish connection we only use `PeerInfo`.
So, I think, they should be separated from each other.